src/ecdsa/ellipticcurve.py

```diff
@@ -186,8 +188,14 @@ def __init__(self, curve, x, y, z, order=None, generator=False):
                 self.__precompute.append((doubler.x(), doubler.y()))
 
     def __getstate__(self):
-        state = self.__dict__.copy()
-        del state['_scale_lock']
+        state = None
```

Comment:
> "When failing to acquire the lock, it now returns None."

I don't think it's possible to fail to acquire lock, it's a blocking operation, and even if it will fail, it will raise an exception, so `return state` won't be reached... as such I don't think `state = None` is necessary

Reply:
Makes sense. Removed.